### PR TITLE
[FLINK-29334][runtime] Remove releaseAndTryRemoveAll in StateHandleStore

### PR DESCRIPTION
**Pull request by WencongLiu**
## What is the purpose of the change

*Remove releaseAndTryRemoveAll in StateHandleStore.*

## Brief change log

  - *Remove releaseAndTryRemoveAll in StateHandleStore.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by WencongLiu** [comment]
@flinkbot run azure

File: flink-runtime/src/test/java/org/apache/flink/runtime/persistence/TestingStateHandleStore.java [hunk 1]
```diff
@@ -121,11 +121,6 @@ public boolean releaseAndTryRemove(String name) throws Exception {
         return removeFunction.apply(name);
     }
 
-    @Override
-    public void releaseAndTryRemoveAll() throws Exception {
-        removeAllRunnable.run();
```
Comment (XComp) [inline flink-runtime/src/test/java/org/apache/flink/runtime/persistence/TestingStateHandleStore.java:126]: `removeAllRunnable` can be removed entirely

**Review by XComp — CHANGES_REQUESTED** [review]
Thanks for working on it, @WencongLiu . There's still some additional cleanup possible, though.

File: flink-runtime/src/test/java/org/apache/flink/runtime/persistence/TestingStateHandleStore.java (hunk 1 above)
Comment (WencongLiu) [reply flink-runtime/src/test/java/org/apache/flink/runtime/persistence/TestingStateHandleStore.java:126]: Done.

Comment (WencongLiu) [reply flink-runtime/src/test/java/org/apache/flink/runtime/persistence/TestingStateHandleStore.java:126]: @XComp Thanks for your review.

**Review by XComp — APPROVED** [review]
LGTM :+1: thanks for your contribution @WencongLiu

**Pull request merged**